snuba/migrations/snuba_migrations/metrics/0001_metrics_sets.py

```diff
@@ -0,0 +1,160 @@
+
+sets_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("metric_id", UInt(64)),
+    Column("granularity", UInt(32)),
+    Column("timestamp", DateTime()),
+    Column("retention_days", UInt(16)),
+    Column("value", AggregateFunction("uniqCombined64", [UInt(64)])),
+]
+
+sets_mv_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("metric_id", UInt(64)),
+    Column("granularity", UInt(32)),
+    Column("timestamp", DateTime()),
+    Column("retention_days", UInt(16)),
+    Column(
+        "tags",
+        Nested([Column("key", Array(UInt(64))), Column("value", Array(UInt(64)))]),
+    ),
+    Column("value", AggregateFunction("uniqCombined64", [UInt(64)])),
+]
+
+
+matview_statement = """
+SELECT
+    org_id,
+    project_id,
+    metric_id,
+    60 as granularity,
+    toStartOfMinute(timestamp) as timestamp,
+    retention_days,
+    tags.key,
+    tags.value,
+    uniqCombined64State(arrayJoin(set_values)) as value
+FROM metrics_buckets_local
+WHERE metric_type = 'set' AND materialization_version = 0
+GROUP BY
+    org_id,
+    project_id,
+    metric_id,
+    tags.key,
+    tags.value,
+    timestamp,
+    granularity,
+    retention_days
+"""
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.METRICS,
+                table_name="metrics_sets_local",
+                columns=sets_columns,
+                engine=table_engines.AggregatingMergeTree(
+                    storage_set=StorageSetKey.METRICS,
+                    order_by="(org_id, project_id, metric_id, granularity, timestamp)",
+                    partition_by="(retention_days, toMonday(timestamp))",
+                    settings={"index_granularity": "256"},
+                ),
+            ),
+            # TODO: Fix the formatting we do in the migrations framework to be able
+            # to add columns called `tags.key` in the CREATE TABLE statement
+            operations.RunSql(
+                storage_set=StorageSetKey.METRICS,
+                statement="""
+                    ALTER TABLE metrics_sets_local ADD COLUMN IF NOT EXISTS `tags.key` Array(UInt64) AFTER retention_days,
+                    MODIFY ORDER BY (org_id, project_id, metric_id, granularity, timestamp, tags.key);
+                """,
+            ),
+            operations.RunSql(
+                storage_set=StorageSetKey.METRICS,
+                statement="""
+                    ALTER TABLE metrics_sets_local ADD COLUMN IF NOT EXISTS `tags.value` Array(UInt64) AFTER tags.key,
+                    MODIFY ORDER BY (org_id, project_id, metric_id, granularity, timestamp, tags.key, tags.value);
+                """,
+            ),
```

Comment:
This nonsense is needed because you cannot create a nested column and reference it in the order by, but we cannot create the individual columns (tas.key, tags.value) because clickhouse wants those columns escaped, but the migraiton framework double escapes them. 
I will see if I can fix the escaping.